datatune/llm/llm.py

```diff
@@ -34,12 +35,55 @@ def _batch_completion(self, prompts: List[str]) -> List[Union[str, Exception]]:
                 ret.append(response["choices"][0]["message"]["content"])
 
         return ret
+    def _true_batch_completion(self, prompts: List[str]) -> List[Union[str,Exception]]:
+        
+        messages = [
+        [{
+            "role": "user",
+            "content": f"{self.prefix}{prompt}"
+        }]
+        for prompt in prompts
+    ]
+        from litellm import batch_completion
+
+        responses = batch_completion(
+            model=self.model_name, messages=messages, **self.kwargs
+        )
+
+        ret = []
+
+        #print(responses)
+        for response in responses:
+            if isinstance(response, Exception):
+                ret.append(response)
+            else:
+                k= response["choices"][0]["message"]["content"].split("<endofresponse>")
+                print(k)
+                print(len(k))
+
+                for i in k:
+                    if i.strip():
+                        ret.append(i.strip()) 
+
+        return ret
 
     def __call__(self, prompt: Union[str, List[str]]) -> List[str]:
         """Always return a list of strings, regardless of input type"""
+
+        self.prefix =(
+            "You will be given multiple questions. Each question will:\n"
+            "- Start with 'Q-[number]:'\n"
+            "- End with '<endofquestion>'\n\n"
+            "You MUST respond to each question in order. For each answer:\n"
+            "- End with '<endofresponse>'\n"
+            "- Do NOT skip or omit any question\n"
+            "Your entire response MUST include one answer per question. Respond strictly in the format described.\n\n"
+            "Questions:\n"
+        )
+        
         if isinstance(prompt, str):
             return [self._completion(prompt)]
-        return self._batch_completion(prompt)
+        return self._true_batch_completion(create_batch_list(prompt, self.model_name, self.prefix))
```

Comment:
```suggestion
        return self._true_batch_completion(prompt)
```
call `create_batch_list` in `_true_batch_completion`